### PR TITLE
feat(playwrighttesting): Added runName feature

### PR DESCRIPTION
**Pull request by kashish2508**
# Contributing to the Azure SDK
Added runName feature
Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: sdk/playwrighttesting/Azure.Developer.MicrosoftPlaywrightTesting.TestLogger/src/Processor/DataProcessor.cs [hunk 1]
```diff
@@ -29,7 +29,7 @@ public TestRunDto GetTestRun()
         {
             var startTime = _cloudRunMetadata.TestRunStartTime.ToString("yyyy-MM-ddTHH:mm:ssZ");
             var gitBasedRunName = ReporterUtils.GetRunName(CiInfoProvider.GetCIInfo())?.Trim();
-            string runName = string.IsNullOrEmpty(gitBasedRunName) ? _cloudRunMetadata.RunId! : gitBasedRunName!;
+            string runName = !string.IsNullOrEmpty(_cloudRunMetadata.RunName) ? _cloudRunMetadata.RunName! : string.IsNullOrEmpty(gitBasedRunName) ? _cloudRunMetadata.RunId! : gitBasedRunName!;
```
Comment (Sid200026) [inline sdk/playwrighttesting/Azure.Developer.MicrosoftPlaywrightTesting.TestLogger/src/Processor/DataProcessor.cs:32]: Inverse the if logic. Also consider moving it to a if else condition for better readability. Also add tests for the same.

**Review by Sid200026 — CHANGES_REQUESTED** [review]

File: sdk/playwrighttesting/Azure.Developer.MicrosoftPlaywrightTesting.TestLogger/src/Constants.cs [hunk 2]
```diff
@@ -36,6 +36,11 @@ public class ServiceEnvironmentVariable
     /// The environment variable for the Playwright service run ID.
     /// </summary>
     public static readonly string PlaywrightServiceRunId = "PLAYWRIGHT_SERVICE_RUN_ID";
+
+    /// <summary>
+    /// The environment variable for the Playwright service run Name.
+    /// </summary>
+    public static readonly string PlaywrightServiceRunName = "PLAYWRIGHT_SERVICE_RUN_NAME";
```
[the remark this answers is not in the document]
Comment (kashish2508) [reply sdk/playwrighttesting/Azure.Developer.MicrosoftPlaywrightTesting.TestLogger/src/Constants.cs:43]: we created a env variable in js/ts package so I thought doing same here, it is not needed will remove.

[the remark this answers is not in the document]
Comment (Sid200026) [reply sdk/playwrighttesting/Azure.Developer.MicrosoftPlaywrightTesting.TestLogger/src/Constants.cs:43]: It's an internal variable - https://github.com/Azure/azure-sdk-for-js/blob/725dc4ca777799339fa3b5afb47f6b48259587b0/sdk/playwrighttesting/microsoft-playwright-testing/src/common/playwrightServiceConfig.ts#L22

File: sdk/playwrighttesting/Azure.Developer.MicrosoftPlaywrightTesting.TestLogger/src/Processor/DataProcessor.cs (hunk 1 above)
Comment (puagarwa) [reply sdk/playwrighttesting/Azure.Developer.MicrosoftPlaywrightTesting.TestLogger/src/Processor/DataProcessor.cs:32]: +1 for increasing readability.
IN nutshell, we will give preference to user passed value when given

**Review by Sid200026 — APPROVED** [review]

**Pull request merged**